BrewMesModulesParent/schedule/src/main/java/com/brewmes/schedule/ScheduleServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.brewmes.schedule;
+
+import com.brewmes.common.entities.ScheduledBatch;
+import com.brewmes.common.services.IScheduleService;
+import com.brewmes.common_repository.ScheduleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class ScheduleServiceImpl implements IScheduleService {
+
+    @Autowired
+    public ScheduleRepository scheduleRepository;
+
+    @Override
+    public int addToQueue(ScheduledBatch scheduledBatch) {
+        ScheduledBatch insertedBatch = scheduleRepository.insert(scheduledBatch);
+        int placement = scheduleRepository.findAll().indexOf(insertedBatch); //NOSONAR
+
+        return placement;
+    }
+
+    @Override
+    public boolean removeFromQueue(String scheduleID) {
+        scheduleRepository.deleteById(scheduleID);
+
+        return !scheduleRepository.existsById(scheduleID);
+    }
+
+    @Override
+    public List<ScheduledBatch> getQueue() {
+
+        return scheduleRepository.findAll();
+    }
+
+    @Override
+    public ScheduledBatch getFirstInQueue() {
```

Comment:
Skal den her metode ikke også fjerne den fra databasen og resortere den?
Jeg antager at den ikke bliver brugt til andet end at få fat i et ScheduleBatch fordi det skal produceres nu.

Reply:
+1

Reply:
Hvis det er intentionen, ville en navneændring måske være på sin plads, bare for at gør det tydeligt. Noget a la `takeFirstInQueue`.

Reply:
+1^